### PR TITLE
update lock

### PR DESCRIPTION
**Pull request by morebtcg**
update lock

### CODE REVIEW

**Comment by bxq2011hust** [comment]
[CodeFactor](https://www.codefactor.io/repository/github/fisco-bcos/fisco-bcos) found an issue: Complex Method

It's currently on:
[libstorage\CachedStorage.cpp:644-814
](https://github.com/FISCO-BCOS/FISCO-BCOS/blob/ca59a185e9450da1ba6095ceb03fe29a076dd396/libstorage/CachedStorage.cpp#L644-L814)Commit ca59a185e9450da1ba6095ceb03fe29a076dd396

**Comment by bxq2011hust** [comment]
[CodeFactor](https://www.codefactor.io/repository/github/fisco-bcos/fisco-bcos) found an issue: Complex Method

It's currently on:
[libstorage\CachedStorage.cpp:221-527
](https://github.com/FISCO-BCOS/FISCO-BCOS/blob/ca59a185e9450da1ba6095ceb03fe29a076dd396/libstorage/CachedStorage.cpp#L221-L527)Commit ca59a185e9450da1ba6095ceb03fe29a076dd396

File: libstorage/CachedStorage.cpp [hunk 1]
```diff
@@ -155,17 +136,15 @@ Entries::Ptr CachedStorage::select(
...
-
     auto out = std::make_shared<Entries>();
 
     auto result = selectNoCondition(hash, num, tableInfo, key, condition);
 
-    Caches::Ptr caches = std::get<0>(result);
+    Cache::Ptr caches = std::get<0>(result);
     auto entries = caches->entriesPtr();
-    for (size_t i = 0; i < entries->size(); ++i)
+    for (auto it : *entries)
```
Comment (wheatli) [inline libstorage/CachedStorage.cpp]: for (auto entry : *entries)

File: libstorage/CachedStorage.cpp [hunk 2]
```diff
@@ -220,7 +198,7 @@ std::tuple<Caches::Ptr, std::shared_ptr<Caches::RWScoped>> CachedStorage::select
 
 size_t CachedStorage::commit(h256 hash, int64_t num, const std::vector<TableData::Ptr>& datas)
 {
-    tbb::spin_mutex::scoped_lock lock(m_cachesMutex);
+    // tbb::spin_mutex::scoped_lock lock(m_cachesMutex);
```
Comment (wheatli) [inline libstorage/CachedStorage.cpp]: just delete it

File: libstorage/Table.cpp [hunk 3]
```diff
@@ -36,31 +36,31 @@
...
-        tbb::spin_mutex::scoped_lock lock(m_data->m_mutex);
-        if (m_data->m_refCount > 0)
-        {
-            --m_data->m_refCount;
-        }
+        --m_data->m_refCount;
     }
 }
 
 uint32_t Entry::getID() const
```
Comment (wheatli) [inline libstorage/Table.cpp]: Set EntryID as uint256

File: libstorage/CachedStorage.h [hunk 4]
```diff
@@ -131,48 +115,53 @@ class CachedStorage : public Storage
...
-    tbb::spin_mutex m_mruMutex;
+    std::shared_ptr<tbb::concurrent_queue<std::tuple<std::string, std::string, ssize_t> > >
+        m_mruQueue;
+
+    tbb::mutex m_commitMutex;
 
     // boost::multi_index
     Storage::Ptr m_backend;
-    size_t m_ID = 1;
+    uint64_t m_ID = 1;
```
Comment (wheatli) [inline libstorage/CachedStorage.h]: Maybe uint256 is more profitable

**Pull request merged**